docs/book/src/tooling/clusterctl.md

```diff
@@ -1 +1,69 @@
-# Clusterctl
+# Using `clusterctl` to create a cluster from scratch
+
+This document provides an overview of how `clusterctl` works and explains how one can use `clusterctl`
+to create a Kubernetes cluster from scratch.
+
+## What is `clusterctl`?
+
+`clusterctl` is a CLI tool to create a Kubernetes cluster. `clusterctl` is provided by the [provider implementations](https://cluster-api.sigs.k8s.io/reference/providers).
+It uses Cluster API provider implementations to provision resources needed by the Kubernetes cluster.
+
+## Creating a cluster
+
+`clusterctl` needs 4 YAML files to start with: `provider-components.yaml`, `cluster.yaml`, `machines.yaml` ,
+`addons.yaml`.
+
+* `provider-components.yaml` contains the *Custom Resource Definitions ([CRDs](https://Kubernetes.io/docs/concepts/extend-Kubernetes/api-extension/custom-resources/))* 
+of all the resources that are managed by Cluster API. Some examples of these resources
+are: `Cluster`, `Machine`, `MachineSet`, etc. For more details about Cluster API resources
+click [here](https://cluster-api.sigs.k8s.io/architecture/controllers).
+* `cluster.yaml` defines an object of the resource type `Cluster`.
+* `machines.yaml` defines an object of the resource type `Machine`. Generally creates the machine
+that becomes the control-plane.
+* `addons.yaml` contains the addons for the provider.
+
+Many providers implementations come with helpful scripts to generate these YAMLS. Provider implementation
+can be found [here](https://cluster-api.sigs.k8s.io/reference/providers).  
+
+`clusterctl` also comes with additional features. For example, `clusterctl` can also take in an optional
+`bootstrap-only-components.yaml` to provide resources to the bootstrap cluster without also providing them
+to the target cluster post-pivot.
+
+For more details about all the supported options run:
+
+```
+clusterctl create cluster --help
+```
+
+After generating the YAML files run the following command:
+
+```
+clusterctl create cluster --bootstrap-type <BOOTSTRAP CLUSTER TYPE> -c cluster.yaml -m machines.yaml -p provider-components.yaml --addon-components addons.yaml
+```
+
+Example usage:
+
+```
+# VMware vSphere
+clusterctl create cluster --provider vsphere --bootstrap-type kind -c cluster.yaml -m machines.yaml -p provider-components.yaml --addon-components addons.yaml
+
+# Amazon AWS
+clusterctl create cluster --provider aws --bootstrap-type kind -c cluster.yaml -m machines.yaml -p provider-components.yaml --addon-components addons.yaml
```

Comment:
We can remove both of the above `--provider` flags, they have been deprecated in v1alpha2

Reply:
Follow-up, we can also remove the two examples and just keep a generic one

Reply:
gotcha. done.